### PR TITLE
Tried - Error Logging for GradientTape #26143

### PR DESCRIPTION
**Pull request by shashvatshahi1998**
i have added isinstance() method to check variable type and raise according exception message

### CODE REVIEW

**Comment by rthadur** [comment]
@shashvatshahi1998 please sign CLA

**Comment by shashvatshahi1998** [comment]
i am trying again and again why this is not working.

**Comment by shashvatshahi1998** [comment]
ok its done

File: tensorflow/python/eager/backprop.py [hunk 1]
```diff
@@ -209,7 +209,9 @@ def grad_fn(*args, **kwds):
     if not variables:
       raise ValueError("No trainable variables were accessed while the "
                        "function was being computed.")
-
+    if not isinstance(variables,float):
```
Comment (alextp) [inline tensorflow/python/eager/backprop.py]: variables is a list of tensorflow variables. You should check their dtypes

**Comment by shashvatshahi1998** [comment]
i have commited once again can you please review it

File: tensorflow/python/keras/activations.py [hunk 2]
```diff
@@ -154,7 +154,7 @@ def relu(x, alpha=0., max_value=None, threshold=0):
   Arguments:
       x: A tensor or variable.
       alpha: A scalar, slope of negative section (default=`0.`).
-      max_value: float. Saturation threshold.
+      max_value: float. Saturation threshold.Returns:
```
Comment (alextp) [inline tensorflow/python/keras/activations.py]: This looks wrong

File: tensorflow/python/eager/backprop.py [hunk 3]
```diff
@@ -209,8 +209,12 @@ def grad_fn(*args, **kwds):
     if not variables:
       raise ValueError("No trainable variables were accessed while the "
                        "function was being computed.")
-
+    for v in variables:  
+      if is_floating(v) == False:
```
Comment (alextp) [inline tensorflow/python/eager/backprop.py]: "if not is_floating(v)" reads better

File: tensorflow/python/eager/backprop.py [hunk 4]
```diff
@@ -209,8 +209,12 @@ def grad_fn(*args, **kwds):
...
+        raise ValueError("Gradient calculation supports only float types.")
```
Comment (alextp) [inline tensorflow/python/eager/backprop.py]: Maybe print v so the user can debug?

File: tensorflow/python/ops/string_ops.py [hunk 5]
```diff
@@ -193,7 +193,7 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
 
 
 @tf_export("string_split")
-def string_split(source, delimiter=" ", skip_empty=True):  # pylint: disable=invalid-name
+def string_split(source, delimiter=" ", skip_empty=True, maxsplit=-1):  # pylint: disable=invalid-name
```
Comment (alextp) [inline tensorflow/python/ops/string_ops.py]: The changes in this file seem unrelated. Adding arguments to public API functions in TF requires an extra review step.

**Review by alextp — CHANGES_REQUESTED** [review]

**Comment by alextp** [comment]
Did you mean to close this PR?

**Pull request closed without merging**